src/components/Global/ButtonIcon.vue

```diff
@@ -15,6 +21,9 @@ export default defineComponent({
     close() {
       return this.icon === "closeMenu" ? true : false;
     },
+    classes() {
+      return [`button__icon ${this.class}`];
+    },
```

Comment:
You could use the fact that Vue gives us attribute inheritance by default, so any class would be added on top of yours on the root element in the template (so our button; especially if you follow the comment I added below).
Also should this class be added on icon or the button? `button__icon` suggests strongly that it would be used on the **icon** but... it's not 👀